sklearn/utils/estimator_checks.py

```diff
@@ -960,6 +960,8 @@ def check_estimators_pickle(name, Estimator):
 
     for method in result:
         unpickled_result = getattr(unpickled_estimator, method)(X)
+        result[method] = np.asarray(result[method]).flatten()
+        unpickled_result = np.asarray(unpickled_result).flatten()
```

Comment:
@vene Sine i could not reproduce the bug locally, I am trying out some alternatives on Travis runs. This is one of them, and I'm ruling it out as a potential solution as it failed. I'll amend my latest commit for the bug to be clear.